packages/Symfony/src/Rector/BinaryOp/ResponseStatusCodeRector.php

```diff
@@ -0,0 +1,186 @@
+        505 => 'HTTP_VERSION_NOT_SUPPORTED',
+        506 => 'HTTP_VARIANT_ALSO_NEGOTIATES_EXPERIMENTAL',
+        507 => 'HTTP_INSUFFICIENT_STORAGE',
+        508 => 'HTTP_LOOP_DETECTED',
+        510 => 'HTTP_NOT_EXTENDED',
+        511 => 'HTTP_NETWORK_AUTHENTICATION_REQUIRED',
+    ];
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Turns status code numbers to constants', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeController
+{
+    public function index()
+    {
+        $response = new \Symfony\Component\HttpFoundation\Response();
+        $response->setStatusCode(200);
+        
+        if ($response->getStatusCode() === 200) {}
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+class SomeController
+{
+    public function index()
+    {
+        $response = new \Symfony\Component\HttpFoundation\Response();
+        $response->setStatusCode(\Symfony\Component\HttpFoundation\Response::HTTP_OK);
+        
+        if ($response->getStatusCode() === \Symfony\Component\HttpFoundation\Response::HTTP_OK) {}
+    }
+}
+CODE_SAMPLE
+
+            )
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Node\Expr\BinaryOp::class, Node\Expr\MethodCall::class];
+    }
+
+    /**
+     * @param Node\Expr\BinaryOp|Node\Expr\MethodCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($node instanceof Node\Expr\MethodCall) {
+            return $this->processMethodCall($node);
+        }
+
+        if ($node instanceof Node\Expr\BinaryOp\Identical || $node instanceof Node\Expr\BinaryOp\NotIdentical || $node instanceof Node\Expr\BinaryOp\Equal || $node instanceof Node\Expr\BinaryOp\NotEqual) {
+            return $this->processBinaryOp($node);
+        }
+
+        return $node;
+    }
+
+    private function processMethodCall(Node\Expr\MethodCall $methodCall): ?Node\Expr\MethodCall
+    {
+        if (!$this->isType($methodCall->var, 'Symfony\\Component\\HttpFoundation\\Response') || !$this->isName($methodCall, 'setStatusCode')) {
+            return null;
+        }
+
+        $statusCode = $methodCall->args[0]->value;
+
+        if (! $statusCode instanceof Node\Scalar\LNumber) {
+            return null;
+        }
+
+        if (!isset($this->codeToConst[$statusCode->value])) {
+            return null;
+        }
+
+        $methodCall->args[0] = new Node\Arg($this->createClassConstant('Symfony\\Component\\HttpFoundation\\Response', $this->codeToConst[$statusCode->value]));
+
+
+        return $methodCall;
+    }
+
+    private function processBinaryOp(Node\Expr\BinaryOp $node): ?Node\Expr\BinaryOp
+    {
+        /** @var Node\Expr\MethodCall $left */
+        $left = $node->left;
+
+        if (!$this->isType($left, 'Symfony\\Component\\HttpFoundation\\Response') || !$this->isName($left, 'getStatusCode')) {
```

Comment:
Here we have the conventions to keep single case conditions:

```php
if (!$this->isType($left, 'Symfony\\Component\\HttpFoundation\\Response')) {
    return null;
} 

if (!$this->isName($left, 'getStatusCode')) {
    return null;
}
```

It makes reading easier and prevents from joining zillions of conditions on one line (or multiple lines).